flag.go

```diff
@@ -62,7 +71,26 @@ func (f *FlagLoader) processField(flagSet *flag.FlagSet, fieldName string, field
 	switch field.Kind() {
 	case reflect.Struct:
 		for _, ff := range field.Fields() {
-			if err := f.processField(flagSet, field.Name()+"-"+ff.Name(), ff); err != nil {
+			flagName := field.Name() + "-" + ff.Name()
+
+			if f.Flatten {
+				// first check if it's set or not, because if we have duplicate
+				// we don't want to break the flag. Just let the user fix it
+				// (by adding the fixme prefix)
+				alreadyDefined := false
+				flagSet.VisitAll(func(fl *flag.Flag) {
+					if strings.ToLower(ff.Name()) == fl.Name {
+						alreadyDefined = true
+					}
+				})
+
+				flagName = ff.Name()
+				if alreadyDefined {
+					flagName = "fixme-" + ff.Name()
```

Comment:
Yeah it's actually panicing when you try to add a duplicate. But we can panic ourself with a better description.